cg/apps/demultiplex/sample_sheet/api.py

```diff
@@ -80,6 +94,78 @@ def validate_sample_sheet(
             file_path=sample_sheet_path, bcl_converter=bcl_converter
         )
 
+    @staticmethod
+    def _is_sample_sheet_from_flow_cell_translatable(flow_cell: FlowCellDirectoryData) -> bool:
+        """
+        Determine if the sample sheet from the flow cell directory is translatable to BCLConvert.
+        """
+        try:
+            flow_cell.run_parameters_path.exists()
+        except FlowCellError:
+            LOG.error(f"Run parameters file for flow cell {flow_cell.full_name} does not exist")
+            return False
+        if not flow_cell.sample_sheet_path.exists():
+            LOG.error(f"Sample sheet for flow cell {flow_cell.full_name} does not exist")
+            return False
+        sample_sheet_content: list[list[str]] = ReadFile.get_content_from_file(
+            file_format=FileFormat.CSV, file_path=flow_cell.sample_sheet_path
+        )
+        if get_sample_type_from_content(sample_sheet_content) is not FlowCellSampleBcl2Fastq:
+            LOG.error(
+                f"Sample sheet for flow cell {flow_cell.full_name} is not a Bcl2Fastq sample sheet"
+            )
+            return False
+        return True
+
+    @staticmethod
+    def _replace_sample_header(sample_sheet_content: list[list[str]]) -> list[list[str]]:
+        """
+        Replace the old sample id header in the Bcl2Fastq sample sheet content with the new one.
+        Raises:
+            SampleSheetError: If the data header is not found in the sample sheet.
+        """
+        for line in sample_sheet_content:
+            if SampleSheetBcl2FastqSections.Data.SAMPLE_INTERNAL_ID_BCL2FASTQ.value in line:
+                idx = line.index(
+                    SampleSheetBcl2FastqSections.Data.SAMPLE_INTERNAL_ID_BCL2FASTQ.value
+                )
+                line[idx] = SampleSheetBCLConvertSections.Data.SAMPLE_INTERNAL_ID.value
+                return sample_sheet_content
+        raise SampleSheetError("Could not find data header in sample sheet")
+
+    def translate_sample_sheet(self, flow_cell_name: str) -> None:
+        """Translate a Bcl2Fastq sample sheet to a BCLConvert sample sheet."""
+        flow_cell: FlowCellDirectoryData = self._get_flow_cell(
+            flow_cell_name=flow_cell_name, bcl_converter=BclConverter.BCLCONVERT
+        )
+        if not self._is_sample_sheet_from_flow_cell_translatable(flow_cell):
+            raise SampleSheetError("Could not translate sample sheet")
+        old_content: list[list[str]] = self._replace_sample_header(
+            ReadFile.get_content_from_file(
+                file_format=FileFormat.CSV, file_path=flow_cell.sample_sheet_path
+            )
+        )
```

Comment:
Why is the old_content retrieved by a function called `_replace_sample_header`? It seems to indicate a change in content

Reply:
fixed